pkg/server/v2/server.go

```diff
@@ -50,28 +47,13 @@ type Server interface {
 
 	// Fetch is the universal fetch method.
 	Fetch(context.Context, *discovery.DiscoveryRequest) (*discovery.DiscoveryResponse, error)
-	// StreamHandler is the universal stream method.
-	StreamHandler(stream Stream, typeURL string) error
```

Comment:
I might be missing it, but without `StreamHandler` defined here `Server` implementation are only required to implement the REST `Server` interface, not the `SotW` interface.  It is harmless at present but seems like an oversight.  It seems you should require both or neither.

Reply:
In reality I would prefer to have neither :) But that breaks ability to use `Server` in `gateway.go`. Probably I can just add sotw.Server and rest.Server here and leave that (otherwise not necessarily needed, IMO) functionality as is.

Reply:
Restored the original behaviour by adding sotw.Server and rest.Server here as well. We can drop that if that is not needed in later PRs. rest.Server (IMO) should be there due to `gateway.go`, while sotw.Server is a different thing - when we add incremental.Server - should we add it there? Cause it requires all other implementors of this server.Server interface to provide it.

Reply:
I wouldn't be opposed. The only thing about incremental is it can't use the same StreamHandler as defined now because it uses a different request/response object: `DeltaDiscoveryRequest` and `DeltaDiscoveryResponse`